doozer/doozerlib/util.py

```diff
@@ -540,84 +540,135 @@ def get_release_name_for_assembly(group_name: str, releases_config: Model, assem
     """
 
     @retry(reraise=True, stop=stop_after_attempt(3), wait=wait_fixed(10))
-    async def _run_command(auth_file: Optional[str]):
-        options = [f'--filter-by-os={go_arch}', '-o', 'json']
+    async def run_oc(auth_file: Optional[str]):
+        opts = ['-o', 'json']
         if auth_file:
-            options.extend([f'--registry-config={auth_file}'])
-        cmd = ['oc', 'image', 'info'] + options + [pull_spec]
+            opts.extend([f'--registry-config={auth_file}'])
+        opts.extend(options)
+        cmd = ['oc', 'image', 'info'] + opts + [pullspec]
         _, out, _ = await exectools.cmd_gather_async(cmd)
         return json.loads(out)
 
+    # If neither a username nor a password were provided, assume an auth file is being used or no auth at all
     if not registry_username and not registry_password:
-        return await _run_command(auth_file=registry_config)
+        return await run_oc(auth_file=registry_config)
 
-    if not registry_password:
-        raise ValueError('registry_password must be provided if auth is needed.')
-    if not registry_username:
-        auth = registry_password
-    else:
+    # Otherwise, a password is always needed for basic auth
+    assert registry_password, 'registry_password must be provided if auth is needed.'
+
+    # Build the basic auth string
+    if registry_username:
         auth = base64.b64encode(f'{registry_username}:{registry_password}'.encode()).decode()
+    else:
+        auth = registry_password
+
+    # Store the basic auth info in a temporary temp file to be used with --registry-config
     with tempfile.NamedTemporaryFile(mode='w', prefix="_doozer_") as registry_config_file:
         registry_config_file.write(json.dumps({
             'auths': {
-                pull_spec.split('/')[0]: {
+                pullspec.split('/')[0]: {
                     'auth': auth,
                 }
             }
         }))
         registry_config_file.flush()
-        return await _run_command(auth_file=registry_config_file.name)
+        return await run_oc(auth_file=registry_config_file.name)
+
+
+async def oc_image_info_for_arch_async(
+        pull_spec: str,
+        go_arch: str = 'amd64',
+        registry_config: Optional[str] = None,
+        registry_username: Optional[str] = None,
+        registry_password: Optional[str] = None,
+) -> Dict:
+    """
+    Runs oc image info with --filter-by-os because images can be multi-arch manifest lists
+    (which cause oc image info to throw an error if not filtered).
+    Will return a single dictionary repesenting the amd64 arch
+    """
+    return await oc_image_info_async(pull_spec, f'--filter-by-os={go_arch}',
+                                     registry_config=registry_config,
+                                     registry_username=registry_username,
+                                     registry_password=registry_password)
+
+
+async def oc_image_info_show_multiarch_async(
+        pull_spec: str,
```

Comment:
nitpik but if I remember we've been using `pullspec` as one word in most places

Reply:
yes, in fact I also renamed some of our old occurrences. let me be more precise with this